### PR TITLE
perf: switch schema validation library

### PR DESCRIPTION
**Pull request by Suor**
voluptuous is compiling schema to a python function, which makes it around 7x faster for me. This is the second slowest part of stage collection, see https://github.com/iterative/dvc/issues/2671.

This only replaces schema validation library for stages, config still uses the old one. There is no point in using both, so if everyone is ok with this switch, I will rewrite the config schema too.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -63,6 +63,8 @@ def run(self):
     "nanotime>=0.5.2",
     "pyasn1>=0.4.1",
     "schema>=0.6.7",
+    "schema>=0.6.7",
```
Comment (efiop) [inline setup.py]: ?

File: dvc/dependency/repo.py [hunk 2]
```diff
@@ -17,11 +16,7 @@ class DependencyREPO(DependencyLOCAL):
     PARAM_REV = "rev"
     PARAM_REV_LOCK = "rev_lock"
 
-    REPO_SCHEMA = {
-        Optional(PARAM_URL): str,
-        Optional(PARAM_REV): str,
-        Optional(PARAM_REV_LOCK): str,
-    }
+    REPO_SCHEMA = {PARAM_URL: str, PARAM_REV: str, PARAM_REV_LOCK: str}
```
Comment (efiop) [inline dvc/dependency/repo.py]: So how does it differentiate required field from an optional one?

Comment (Suor) [reply dvc/dependency/repo.py]: There is a `Required()` wrapper.

**Comment by Suor** [comment]
Switched config validation to voluptuous too, removed the req for schema.

**Comment by efiop** [comment]
@Suor Please rebase.

File: setup.py [hunk 3]
```diff
@@ -62,7 +62,7 @@ def run(self):
     "setuptools>=34.0.0",
     "nanotime>=0.5.2",
     "pyasn1>=0.4.1",
-    "schema>=0.6.7",
+    "voluptuous>=0.11.7",
```
Comment (efiop) [inline setup.py]: Need to self: need to not forget to update it in conda package deps.

**Pull request merged**